### PR TITLE
Document GLAM datasets

### PR DESCRIPTION
**Pull request by robhudson**

### CODE REVIEW

File: src/datasets/glam.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# GLAM datasets
+
+## Introduction
+
+The GLAM aggregation tables are useful for getting the SQL view of the data that drives GLAM if more exploration is required. GLAM aims to answer a majority of the "easy" questions of how a probe or metric has changed over time. Exploring the GLAM tables could take that a little farther. If you need to dive deeper or aggregate on a field that isn't included here, consider reading [todo].
```
Comment (wlach) [inline src/datasets/glam.md:5]: Long paragraphs are a little easier to read / edit if you separate them out per line (I originally found this weird but have become addicted to it). Also I think you missed a reference.

```suggestion
[GLAM](https://glam.telemetry.mozilla.org) aims to answer a majority of the "easy" questions of how a probe or metric has changed over time. 
The GLAM  aggregation tables are useful for accessing the data that drives GLAM if more exploration is required.
Exploring the GLAM tables could take that a little farther, but still has some limitations. 
If you need to dive deeper or aggregate on a field that isn't included here, consider reading [Visualizing Percentiles of a Main Ping Exponential Histogram](https://docs.telemetry.mozilla.org/cookbooks/main_ping_exponential_histograms.html).
```

File: src/datasets/glam.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+The GLAM tables:
+* Are aggregated at the client level, not the submission ping level
+* Provide a set of dimensions for subsets: channel, os, process or ping type
```
Comment (wlach) [inline src/datasets/glam.md:9]:
```suggestion
* Provide a set of dimensions for subsets: channel, OS, process or ping type
```

File: src/datasets/glam.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    * `metric`: The name of the metric
+    * `metric_type`: The type of metric, e.g. "histogram-enumerated"
+    * `key`: The key if the metric is a keyed metric
+    * `process`: The process
+    * `client_agg_type`: The type of client aggregation used, e.g. "summed_histogram"
+    * `agg_type`: One of histogram or percentiles representing what data will be in the `aggregates` column
+    * `total_users`: The number of users that submitted data for the combination of dimensions
+    * `aggregates`: The data as a key/value record, either percentiles or histogram
+
+#### Sample query
```
Comment (wlach) [inline src/datasets/glam.md:40]:
```suggestion

#### Sample query
```

File: src/datasets/glam.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  `moz-fx-data-shared-prod.telemetry.client_probe_counts`
+WHERE
+  metric="checkerboard_severity"
+  AND channel="nightly"
+  AND os IS NULL
+  AND process="parent"
+  AND app_build_id IS NULL
+```
+
+NOTE:
```
Comment (wlach) [inline src/datasets/glam.md:65]:
```suggestion
Notes:
```

File: src/datasets/glam.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+* To query all OSes, use: `WHERE os IS NULL`
+* To query by build ID, use: `WHERE app_build_id IS NOT NULL`
+* To query by version, use: `WHERE app_build_id IS NULL`
+
+### Firefox for Android
+
+* Data source tables
```
Comment (wlach) [inline src/datasets/glam.md:72]:
```suggestion
#### Data source tables
```

File: src/datasets/glam.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    * `org_mozilla_fenix_glam_release__view_probe_counts_v1`
+    * `org_mozilla_fenix_glam_beta__view_probe_counts_v1`
+    * `org_mozilla_fenix_glam_nightly__view_probe_counts_v1`
+* Data reference
```
Comment (wlach) [inline src/datasets/glam.md:76]:
```suggestion

#### Data reference
```

File: src/datasets/glam.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+  mozfun.glam.histogram_cast_json(aggregates) AS aggregates,
+FROM
+  `moz-fx-data-shared-prod.glam_etl.org_mozilla_fenix_glam_release__view_probe_counts_v1`
+WHERE
+  metric="performance_time_dom_complete"
+  AND os="Android"
+  AND ping_type="*"
+  AND app_build_id!="*"
+```
+NOTE:
```
Comment (wlach) [inline src/datasets/glam.md:112]:
```suggestion

Notes:
```

File: src/datasets/glam.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+### Firefox Desktop
+
+* Optional info?
+    * Pre-processed from Main telemetry to intermediate data with one row per client per metric per day, then aggregated normalizing across clients.
+    * Clients daily aggregates analagous to clients daily with:
+        * all metrics aggregated
+        * each scalar includes min, max, average, sum, and count aggregations
+        * each histogram aggregated over all client data per day
+        * each date is further aggregated over the dimensions: channel, os, version, build ID
```
Comment (wlach) [inline src/datasets/glam.md:21]: Could you turn this into a paragraph with complete sentences? I think that'll flow a little better.

File: src/datasets/glam.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+
+* Data source table
```
Comment (wlach) [inline src/datasets/glam.md:24]:
```suggestion
#### Data source table
```

File: src/datasets/glam.md [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+    * `moz-fx-data-shared-prod.telemetry.client_probe_counts`
+* Data reference
```
Comment (wlach) [inline src/datasets/glam.md:26]:
```suggestion
#### Data reference
```

File: src/datasets/glam.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+  AND process="parent"
+  AND app_build_id IS NULL
+```
+
+NOTE:
+* To query all OSes, use: `WHERE os IS NULL`
+* To query by build ID, use: `WHERE app_build_id IS NOT NULL`
+* To query by version, use: `WHERE app_build_id IS NULL`
+
+### Firefox for Android
```
Comment (wlach) [inline src/datasets/glam.md:70]: I'm hoping to autogenerate docs for tables that derive from a Glean Application in the Glean Dictionary soon (some of what's here will be very useful for that!)

File: src/datasets/glam.md [hunk 12]
```diff
@@ -0,0 +1,115 @@
+# GLAM datasets
+
+## Introduction
+
```
Comment (wlach) [inline src/datasets/glam.md:4]: Can probably skip the introduction heading-- it's implied

```suggestion
```

**Review by wlach — COMMENTED** [review]
Some preliminary comments. I'd lean towards landing this initial version sooner than later, then incrementally updating it. This is already very useful IMO.

File: src/datasets/glam.md (hunk 8 above)
Comment (robhudson) [reply src/datasets/glam.md:21]: This section needs some work. I wanted to share some details on the intermediate tables but I personally have a bit limited knowledge of them, so I took some notes of what I was able to find. And added the "optional" just for me to follow up on that.

**Review by wlach — APPROVED** [review]
Looks good! Thanks for putting this together

**Pull request merged**